test/test_torch.py

```diff
@@ -3776,12 +3768,55 @@ def test_trilu_indices(self):
         self.assertEqual(
             x.triu(0).nonzero().transpose(0, 1), torch.triu_indices(3, 3))
 
-    def test_triu(self):
-        x = torch.rand(SIZE, SIZE)
-        res1 = torch.triu(x)
-        res2 = torch.Tensor()
-        torch.triu(x, out=res2)
-        self.assertEqual(res1, res2, 0)
+    @staticmethod
+    def _test_triu_tril(self, cast):
```

Comment:
Could you also add some tests for non-contiguous-inputs to `torch.triu, tensor.triu_, torch.tril, and tensor.tril_`?

Reply:
Yes, I will get this done. Thank you for the suggestions.